Firestore/Source/Model/FSTFieldValue.h

```diff
@@ -288,4 +279,11 @@ enum class ServerTimestampBehavior { None, Estimate, Previous };
 
 @end
 
+/**
+ * A value that delegates to the c++ model::FieldValue.
+ */
+@interface FSTDelegateValue<__covariant T> : FSTFieldValue <T>
++ (instancetype)delegateValue:(FieldValue &&)value;
```

Comment:
Objective-C method names follow a pattern where you end up with

  * noun describing result
  * a preposition
  * label describing the argument

That would make this `delegateWithValue:` or something like that.

You could side-step the apparent verbosity of this slightly by adding a method like this to `FieldValue`:

```c++
#if __OBJC__
FSTFieldValue* Wrap() && {
  return [FSTDelegateValue delegateWithValue:std::move(*this)];
}
#endif  // __OBJC__
```

Then callers see

```c++
FieldValue::True().Wrap()
```
Instead of
```
[FSTDelegateValue delegateValue:FieldValue::True()]
```